node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -1367,6 +1470,213 @@ mod tests {
+        );
+        let stream_handler_pool_recording = stream_handler_pool_recording_arc.lock().unwrap();
+        assert_eq!(
+            stream_handler_pool_recording.get_record::<RemoveStreamMsg>(0),
+            &RemoveStreamMsg {
+                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
+                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[0]),
+                stream_type: RemovedStreamType::Clandestine,
+                sub: dispatcher_sub.clone()
+            }
+        );
+        assert_eq!(
+            stream_handler_pool_recording.get_record::<RemoveStreamMsg>(1),
+            &RemoveStreamMsg {
+                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
+                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[1]),
+                stream_type: RemovedStreamType::Clandestine,
+                sub: dispatcher_sub
+            }
+        );
+    }
+
+    #[test]
+    fn properly_constructed_originate_only_ipchange_is_identified_and_handled() {
+        let old_addr = NodeAddr::new (&IpAddr::from_str("1.2.3.4").unwrap(), &vec![2000, 2001]);
+        let new_addr = NodeAddr::new (&IpAddr::from_str("2.3.4.5").unwrap(), &vec![2000, 2001]);
+        let gossip_source = SocketAddr::from (new_addr.clone());
+        let (gossip, new_node, mut db) =
+            make_ipchange (2345, Some (old_addr.clone()), Some (new_addr),
+                           Mode::OriginateOnly);
+        let cryptde = CryptDENull::from(db.root().public_key(), TEST_DEFAULT_CHAIN);
+        let agrs_vec: Vec<AccessibleGossipRecord> = gossip.try_into().unwrap();
+        let (cpm_recipient, _) = make_cpm_recipient();
+        let (stream_handler_pool, _, stream_handler_pool_recording_arc) = make_recorder();
+        let (dispatcher, _, _) = make_recorder();
+        let stream_handler_pool_sub = stream_handler_pool.start().recipient();
+        let dispatcher_sub = dispatcher.start().recipient();
+        let subject = IpChangeHandler::new(stream_handler_pool_sub.clone(),
+            dispatcher_sub);
+        let system = System::new("test");
+
+        let qualifies_result =
+            subject.qualifies(&db, &agrs_vec.as_slice(), gossip_source);
+        let handle_result = subject.handle(
+            &cryptde,
+            &mut db,
+            agrs_vec,
+            gossip_source,
+            &vec![],
+            &cpm_recipient,
+        );
+
+        System::current().stop();
+        system.run();
+        assert_eq!(Qualification::Matched, qualifies_result);
+        assert_eq!(
+            handle_result,
+            GossipAcceptanceResult::Absorbed,
+        );
+        assert_eq!(
+            db.node_by_key(new_node.public_key()).unwrap().node_addr_opt(),
+            None
+        );
+        let stream_handler_pool_recording = stream_handler_pool_recording_arc.lock().unwrap();
+        assert_eq! (stream_handler_pool_recording.len(), 0);
```

Comment:
I'm not having a bright mind...but thinking, why wouldn't we discard useless streams for an originate-only Node if we've had to have some. I understand that after the discovery in the dev call this message becomes pointless in overall but it leaves me with an impression this part of this test was wrong.

Reply:
Yeah, that's a good idea. We have a stream open to the new IP address (obviously: that's the stream the IpChange Gossip came over), but it'd still be a good idea to get rid of the others so that we don't try to use them. I'll work on it.

Reply:
No. There isn't any "ping" transaction you can send down a TCP stream the way there is on a WebSockets connection. We could invent one, but I don't want to, at least not on this card.

I'll make another card, though.